src/features/tokenGatedRooms/sagas/join.saga.ts

```diff
@@ -0,0 +1,69 @@
+import { TokenGatedRoomAction } from '$/features/tokenGatedRooms'
+import { takeEvery } from 'redux-saga/effects'
+import handleError from '$/utils/handleError'
+import { JoinPolicyRegistryAddress } from '$/features/tokenGatedRooms/utils/const'
+
+import { Contract, providers } from 'ethers'
+
+import * as JoinPolicyRegistry from '$/contracts/JoinPolicyRegistry.sol/JoinPolicyRegistry.json'
```

Comment:
Maybe `*` in this file could be scoped better? E.g.
```ts
import { abi as jprAbi } from '…'
```

Reply:
Makes sense, replaced it on the last commit